### ISSUE TITLE
modisom 3.0.0 incompatible with kbmag

### ISSUE THREAD

**Issue by fingolfin** [gap-packages/modisom#20]
Because of this in modisom 3.0.0:
```
gap/grpalg/collect.gi
47:BindGlobal("ReducedWord", function( w )
```

Note that `kbmag` defines an *operation* `ReducedWord` and has done so for decades.

Possible fixes:
- if it is just internal (which I guess it is, as it is undocumented), simply rename it to e.g. `MODISOM_ReducedWord`
- if you plan to expose it to the user in the long run, you could turn yours into an operation, too. It is valid for multiple packages to declare operations with the same name, as long as the methods are defined for different inputs.

I also recommend testing whether your package works when you do `LoadAllPackages`. (I just tried this for you and it seems the above is the only such case).

Let me know if fix 1 is OK for you, then I can quickly apply it and re-release.

**Comment by margollo** [comment]
Hi Max,

Thanks for noticing. Indeed, the function is only internal. Renaming it seems the best and easiest solution. Please go ahead and change it.

All the best,
Leo
________________________________
From: Max Horn ***@***.***>
Sent: Monday, September 23, 2024 10:46 AM
To: gap-packages/modisom ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [gap-packages/modisom] modisom 3.0.0 incompatible with kbmag (Issue #20)


Because of this in modisom 3.0.0:

gap/grpalg/collect.gi
47:BindGlobal("ReducedWord", function( w )


Note that kbmag defines an operation ReducedWord and has done so for decades.

Possible fixes:

  *   if it is just internal (which I guess it is, as it is undocumented), simply rename it to e.g. MODISOM_ReducedWord
  *   if you plan to expose it to the user in the long run, you could turn yours into an operation, too. It is valid for multiple packages to declare operations with the same name, as long as the methods are defined for different inputs.

I also recommend testing whether your package works when you do LoadAllPackages. (I just tried this for you and it seems the above is the only such case).

Let me know if fix 1 is OK for you, then I can quickly apply it and re-release.

**Comment by fingolfin** [comment]
This was resolved in 41ed066a5ce7b1141fd2e43eaa8371b37671a1d4
